### ISSUE TITLE
schema mismatch in Parquet glob when casting INTEGER to BIGINT is harmless

### ISSUE THREAD

**Issue by zozotintin** [duckdb/duckdb#1214]
Some SQL query engines like [Apache Drill](https://drill.apache.org/) creates multiple `Parquet` partitions for a single table. If a table contains a `BIGINT` column, but some `Parquet` partitions happen to have small numbers, those partitions will have `INTEGER` column instead to save space. 

So when I read the partitions with `duckdb`'s `parquet_scan` glob, it complains
```
schema mismatch in Parquet glob: column "id" in parquet file is of type BIGINT, but in the original file "" this column is of type "INTEGER"
```

I hope `duckdb` can automatically detect these kinds of harmless schema mismatch problems, and cast `INTEGER` to `BIGINT` when necessary.

Thank you very much.

**Comment by hannesmuehleisen** [comment]
The issue here is that we need to decide quite early on which type we should assume for an input column, during the so-call binding phase. When reading multiple files at the same time, we currently only read the metadata of the first file due to performance considerations.

 In order to do this correctly with multiple Parquet files, we could either read all the metadata of all files covered by the filename glob, which will be very slow, or auto-cast the values if they are different from what was assumed from the first file. This will work fine if the values fit, but from what you describe this explicitly happens because values are bigger than what was declared in the first file. 

Both solutions seem problematic to me unfortunately. Is there a setting to make Drill be consistent there?

**Comment by zozotintin** [comment]
Thank you for clarifying the technical difficulties to me. I thought reading the metadata is a lightweight operation because the metadata's size is small, but perhaps I was wrong. `Drill` is able to handle the mismatch probably because it adopts a slow approach. I will check whether I can set `Drill` to be consistent in data types when exporting tables.

**Comment by hannes** [comment]
Should be fixed in #3238
